layout.py

```diff
@@ -16,6 +16,12 @@ def create_app():
         Dash: A Dash web application instance.
     """
 
+    # This section redundantly reads from 'config.json' due to limitations in accessing configuration data.
+    # It's not feasible to use 'get_config_data()' from 'app.py' because it relies on flask-caching
+    # for memoization, which can only be set up after the 'app' object instantiation.
+    # Additionally, 'url_base_pathname' can only be set during app instantiation, not afterwards.
+    # Thus, we cannot instantiate 'app', set up flask-caching, fetch config data using the memoized
+    # 'get_config_data()', and then update 'url_base_pathname'.
```

Comment:
See this thread where I am the OP: https://community.plotly.com/t/what-is-the-proper-way-to-handle-user-defined-app-configuration-options-from-a-file-in-plotly-dash/83216

jinnyzor said:

> The whole app is in stateless form, this is true

Conventional Plotly Dash code wants to completely avoid global variables because everything is supposed to be stateless, functions are supposed to be pure, highly testable code, and all that jazz, and the replacement for it that they want you to do is to either:

1. Use dcc.Store to store global data in the browser, but visible to client
2. Use `flask-caching`, to store data server-side, either in a redis cache, memory, filesystem, etc.

Reply:
So when they said serverSide store they meant flask-caching?

Reply:
> So when they said serverSide store they meant flask-caching?

I believe that is along the lines of what they meant, based on their organization's documentation: https://dash.plotly.com/sharing-data-between-callbacks#storing-shared-data

> While avoiding global state is certainly a good idea in general it should not be taken as dogma

I've always felt like the term "stateless" is a bit of an exaggeration, and, realistically, would be more accurate if it was called "state-avoidant", but that wouldn't sound as sexy. But, since I saw Plotly's dogma in their official documentation and forum responses, I decided to just trust their authority on the matter. I suppose the sign of a thoughtful developer would be to know the special cases where the 100% literal interpretation of blanket statements such as "stateless" start becoming a hassle instead of being practical.

I just pushed a commit that addresses the complaint in this thread. The app works and all tests pass when ran locally.